### PR TITLE
Restore Reach-based Lakes from v5.1.x

### PR DESCRIPTION
**Pull request by rcabell**
TYPE: bug fix

KEYWORDS: reach, lakes, routing

SOURCE: NCAR

DESCRIPTION OF CHANGES:  support for lakes (reservoirs) in non-UDMP reach-based routing was added in version 5.1.1 but subsequently dropped from subsequent releases. This PR adds that functionality back in.

TESTS CONDUCTED: testing offline, will update in comments if necessary

### CODE REVIEW

File: src/Routing/module_channel_routing.F90 [hunk 1]
```diff
@@ -801,29 +831,39 @@ Subroutine drive_CHANNEL(did, latval,lonval,KT, IXRT,JXRT, SUBRTSWCRT, &
...
 
-                   else
+                else
                     print *, "FATAL ERROR: no channel option selected"
                     call hydro_stop("In drive_CHANNEL() - no channel option selected")
                    endif
 !            endif !!! order(1) .ne. 1
          end do       !--k links
 
+   #ifdef MPP_LAND
```
Comment (scrasmussen) [inline src/Routing/module_channel_routing.F90:861]: These `#ifdef` and `#endif` preprocessor statements are indented and are giving my compiler issues. Removing the leading spaces should fix the issue

**Review by scrasmussen — CHANGES_REQUESTED** [review]
Looks good! Double checked the v5.1.1 code and looks like all the needed changes were brought over but with improved formatting 👍

File: src/Routing/module_channel_routing.F90 (hunk 1 above)
Comment (rcabell) [reply src/Routing/module_channel_routing.F90]: Whoops! Not sure how that happened - Intel Fortran didn’t seem to care. Fixed now…

**Review by scrasmussen — DISMISSED** [review]
That fix looks good and now the CI is passing (always comforting!)

**Review by scrasmussen — APPROVED** [review]
LGTM!

**Pull request merged**